test/controllers/playlists_controller_test.rb

```diff
@@ -0,0 +1,42 @@
+require "test_helper"
+
+class PlaylistsControllerTest < ActionDispatch::IntegrationTest
+
+  setup do
+    @playlist = playlists(:one)
+  end
+
+  test "can get index" do
+    get playlists_url
+    assert_response :success
+  end
+
+  test "can view playlist" do
+    get playlist_url(@playlist)
+    assert_response :success
+  end
+
+  test "can get new" do
+    get new_playlist_url
+    assert_response :success
+  end
+
+  test "can create a playlist" do
+    assert_difference('Playlist.count') do
+      post playlists_url, params: { playlist: {name: "Playlist1", energy: "60", danceability: "65", genre: "genre1"} }
+    end
+    assert_redirected_to playlist_url(Playlist.last)
+  end
```

Comment:
Après avoir relu de la documentation, je me rends compte que je confondais system tests et functionnal tests par rapport à comment c'était présenté dans la doc de rails (cf https://guides.rubyonrails.org/testing.html#functional-and-system-testing ).

En pratique, je vois surtout des tests systèmes que des tests de controller/fonctionnels, et personnellement c'est ce que je fais aussi.

Tes tests sont clairement valides, mais je t'invite à faire un test "système" comme présenté dans la doc, en testant la création de playlist en partant de la page new (comme ici: https://guides.rubyonrails.org/testing.html#creating-articles-system-test par exemple).

Reply:
L'avantage de ces tests c'est que ça se lit bien et que ça simule vraiment le comportement user.